File: app/src/main/java/com/pashkobohdan/ttsreader/TTSReaderProApplication.java
package com.pashkobohdan.ttsreader;

import android.app.Application;

import com.pashkobohdan.ttsreader.di.ApplicationComponent;
import com.pashkobohdan.ttsreader.di.DaggerApplicationComponent;
import com.pashkobohdan.ttsreader.di.modules.AppModule;
import com.pashkobohdan.ttsreader.di.modules.DatabaseModule;

public class TTSReaderProApplication extends Application {

    public static TTSReaderProApplication INSTANCE;
    private ApplicationComponent component;

    @Override
    public void onCreate() {
        super.onCreate();
        INSTANCE = this;
        //TODO maybe run first getApplicationComponent() here
    }


    public ApplicationComponent getApplicationComponent() {
        if (component == null) {
            component = DaggerApplicationComponent
                    .builder()
                    .appModule(new AppModule(this))
                    .databaseModule(new DatabaseModule(this))
                    .build();
        }
        return component;
    }
}

File: app/src/pro/java/com/pashkobohdan/ttsreader/pro/TTSReaderProApplication.kt
package com.pashkobohdan.ttsreader.pro

import com.pashkobohdan.ttsreader.TTSReaderApplication
import com.pashkobohdan.ttsreader.di.ApplicationComponent
import com.pashkobohdan.ttsreader.di.DaggerApplicationComponent
import com.pashkobohdan.ttsreader.di.modules.AppModule
import com.pashkobohdan.ttsreader.di.modules.DatabaseModule
import com.pashkobohdan.ttsreader.pro.di.modules.ProSettingsModule

class TTSReaderProApplication : TTSReaderApplication() {

    override fun getApplicationComponent(): ApplicationComponent {
        if (component == null) {
            component = DaggerApplicationComponent
                    .builder()
                    .appModule(AppModule(this))
                    .databaseModule(DatabaseModule(this))
                    .settingsModule(ProSettingsModule())
                    .build()
        }
        return component!!
    }
}
